omod/src/main/java/org/openmrs/module/adminui/page/controller/account/ChangePasswordPageController.java

```diff
@@ -59,49 +63,104 @@ public String post(@MethodParam("getChangePassword") @BindParams ChangePassword
 		if (errors.hasErrors()) {
 			sendErrorMessage(errors, messageSource, request);
 			model.addAttribute("errors", errors);
+			setModelForView(model);
 			return "account/changePassword";
 		} else {
-			return changePasswords(changePassword, userService, messageSourceService, request);
+			return changePasswords(changePassword, userService, messageSourceService, request, model);
 		}
 	}
 	
 	private String changePasswords(ChangePassword changePassword, UserService userService,
-	                               MessageSourceService messageSourceService, HttpServletRequest request) {
+	                               MessageSourceService messageSourceService, HttpServletRequest request, PageModel model) {
 		try {
 			userService.changePassword(changePassword.getOldPassword(), changePassword.getNewPassword());
 			request.getSession().setAttribute(AdminUiConstants.SESSION_ATTRIBUTE_INFO_MESSAGE,
 			    messageSourceService.getMessage("emr.account.changePassword.success", null, Context.getLocale()));
 			request.getSession().setAttribute(AdminUiConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
+			return "account/myAccount";
 		}
 		catch (DAOException e) {
 			request.getSession().setAttribute(
 			    AdminUiConstants.SESSION_ATTRIBUTE_ERROR_MESSAGE,
 			    messageSourceService.getMessage("adminui.account.changePassword.fail", new Object[] { e.getMessage() },
 			        Context.getLocale()));
+			setModelForView(model);
 			return "account/changePassword";
 		}
 		
-		return "account/myAccount";
+
 	}
-	
+
+	/**
+	 * Uses the following security settings to validate new password
+	 * security.passwordCannotMatchUsername
+	 * security.passwordCustomRegex
+	 * security.passwordMinimumLength
+	 * security.passwordRequiresDigit
+	 * security.passwordRequiresNonDigit
+	 * security.passwordRequiresUpperAndLowerCase
+	 * @return
+	 */
 	private void validatePasswords(ChangePassword changePassword, BindingResult errors,
 	                               MessageSourceService messageSourceService) {
+
+		//load global properties
+		boolean GP_PASSWORD_CANNOT_MATCH_USERNAME_OR_SYSTEMID = Boolean.parseBoolean(Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GP_PASSWORD_CANNOT_MATCH_USERNAME_OR_SYSTEMID));
```

Comment:
There is a getGlobalProperty that lets you pass in the type of the returned value, so that you don't have to do call parseBoolean